### ISSUE TITLE
Feature Request:  Serialize IPAddress

### ISSUE THREAD

**Issue by EmperorArthur** [bblanchon/ArduinoJson#1444]
Currently compilation fails if ArdunioJson attempts to serialize an IPAddress object.

On the surface, this makes sense because it is not something which can be directly converted to JSON.  However, Serial.print(..) explicitly works, so it catches users off gaurd.

Code to convert IPAddress to a string, which can be serialized.
```
out = String(address[0]) + "." + String(address[1]) + "." + String(address[2]) + "." + String(address[3])
```

**Comment by bblanchon** [comment]
We can make this feature more general by including any `Printable` type.

**Comment by EmperorArthur** [comment]
That would be great!

Hmm, Printable types implement `printTo(Print& p) const`.  The good news, is that it looks like you can get away with just implementing `size_t write(uint8_t c)`, on a subclass, though you could also overwrite `size_t write(const uint8_t *buffer, size_t size)` for more efficient operation.  Then again, I'm speaking to someone who has much greater experience with the code than I do, as your "PrintWriter" shows.

**Comment by bblanchon** [comment]
Feature published in [ArduinoJson 6.18.0](https://arduinojson.org/news/2021/05/04/version-6-18-0/)
